src/fiatconnect/FiatDetailsScreen.tsx

```diff
@@ -0,0 +1,330 @@
+
+  return newSchema
+}
+
+const FiatDetailsScreen = ({ route, navigation }: Props) => {
+  const { t } = useTranslation()
+  const { providerURL, fiatAccountSchema, cicoQuote } = route.params
+  const [validInputs, setValidInputs] = useState(false)
+  const [textValue, setTextValue] = useState('')
+  const [errors, setErrors] = useState(new Set<string>())
+  const inputRefs = useRef<string[]>([textValue])
+  const userCountry = useSelector(userLocationDataSelector)
+
+  const dispatch = useDispatch()
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      headerTitle: i18n.t('fiatDetailsScreen.header'),
+    })
+  }, [navigation])
+
+  const getFieldsBySchema = (fiatAccountSchema: FiatAccountSchema): Fields[] => {
+    return SCHEMA_TO_FIELD_METADATA_MAP[fiatAccountSchema]
+  }
+
+  const formFields = useMemo(() => {
+    const fields = getFieldsBySchema(fiatAccountSchema)
+
+    for (let i = 0; i < fields.length; i++) {
+      inputRefs.current.push('')
+    }
+    return fields
+  }, [fiatAccountSchema])
+
+  const onPressNext = async () => {
+    validateInput()
+
+    if (validInputs) {
+      const body: Record<string, string> = {}
+      for (let i = 0; i < formFields.length; i++) {
+        body[formFields[i].name] = inputRefs.current[i]
+      }
+
+      const validatedBody = validateAndCompleteSchema(
+        body,
+        getSchemaObjectByType(fiatAccountSchema)
+      )
+
+      await addNewFiatAccount(providerURL, fiatAccountSchema, validatedBody)
+        .then((data) => {
+          // TODO Tracking here
+          dispatch(showMessage(t('fiatDetailsScreen.addFiatAccountSuccess')))
+        })
+        .catch((error) => {
+          // TODO Tracking here
+          if (error === FiatConnectError.ResourceExists) {
+            dispatch(showError(ErrorMessages.ADD_FIAT_ACCOUNT_RESOURCE_EXIST))
+          } else {
+            dispatch(showError(t('fiatDetailsScreen.addFiatAccountFailed')))
+          }
+          Logger.error(TAG, `Error adding fiat account: ${error}`)
+        })
+
+      // TODO: navigate to the next screen
+    }
+  }
+
+  const validateAndCompleteSchema = (
+    body: Record<string, string>,
+    schemaObject: AccountNumber | undefined
+  ): Record<string, string> | undefined => {
+    if (!schemaObject) {
+      Logger.error(TAG, 'Cannot create a schema object, check the schema passed from the Prop')
+      return
+    }
+
+    for (const [key] of Object.entries(schemaObject)) {
+      if (!body[key]) {
+        if (key === 'country') {
+          if (!userCountry) {
+            Logger.error(TAG, 'User country is not available from redux')
+            return
+          }
+          body[key] = userCountry.countryCodeAlpha2 || ''
+        } else if (key === 'fiatAccountType') {
+          body[key] = FiatAccountType.BankAccount
+        }
+      }
+    }
+    return body
+  }
+
+  const onPressSelectedPaymentOption = () => {
+    // TODO: tracking here
+    // TODO: navigate to SelectProvider screen
```

Comment:
This just goes to this screen right? https://github.com/valora-inc/wallet/blob/main/src/fiatExchanges/SelectProvider.tsx

I don't think the props for that will change much, so i I think it can safely be implemented now